lib/web/ui/cluster.go

```diff
@@ -20,33 +20,55 @@ import (
 	"sort"
 	"time"
 
+	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
 )
 
 // Cluster describes a cluster
 type Cluster struct {
 	// Name is the cluster name
 	Name string `json:"name"`
 	// LastConnected is the cluster last connected time
-	LastConnected time.Time `json:"last_connected"`
+	LastConnected time.Time `json:"lastConnected"`
 	// Status is the cluster status
 	Status string `json:"status"`
+	// NodeCount is this cluster number of registered servers
+	NodeCount int `json:"nodeCount"`
+	// PublicURL is this cluster public URL (its first available proxy URL)
+	PublicURL string `json:"publicURL"`
 }
 
-// NewClusters creates a slice of Clusters
-func NewClusters(remoteClusters []reversetunnel.RemoteSite) []Cluster {
+// NewClusters creates a slice of Cluster's, containing data about each cluster.
+func NewClusters(remoteClusters []reversetunnel.RemoteSite) ([]Cluster, error) {
 	clusters := []Cluster{}
-	for _, item := range remoteClusters {
+	for _, rclsr := range remoteClusters {
+		clt, err := rclsr.CachingAccessPoint()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		// error is not handled b/c nil proxies means
+		// proxyHost will be set to an empty value
+		proxies, _ := clt.GetProxies()
```

Comment:
Lets explicitly handle this error by doing this

```golang
proxies, err := clt.GetProxies()
if err != nil {
  proxies = []Cluster{}
}
```

Reply:
And lets log the error. This logging is legit because it's really an issue when it returns an error. There was another logging message which was ok to remove -> _log.Debugf("public_address not set for proxy, returning proxyHost: %q", proxyHost)_

Reply:
revised.

I didn't think `proxies = []Cluster{}` was necessary, b/c `GuessProxyHost` will check for length first which `len(nil)` is zero. B/c my thinking is you will always want to check for length before accessing any indexes. But I placed it in anyways :)